test/e2e/placement_negative_cases_test.go

```diff
@@ -0,0 +1,223 @@
+					Name: crpName,
+					// Add a custom finalizer; this would allow us to better observe
+					// the behavior of the controllers.
+					Finalizers: []string{customDeletionBlockerFinalizer},
+				},
+				Spec: placementv1beta1.ClusterResourcePlacementSpec{
+					ResourceSelectors: workResourceSelector(),
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickFixedPlacementType,
+						ClusterNames: []string{
+							memberCluster1EastProdName,
+						},
+					},
+					Strategy: placementv1beta1.RolloutStrategy{
+						Type: placementv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+							UnavailablePeriodSeconds: ptr.To(2),
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			Eventually(func() error {
+				crp := &placementv1beta1.ClusterResourcePlacement{}
+				if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+					return err
+				}
+
+				wantStatus := placementv1beta1.ClusterResourcePlacementStatus{
+					Conditions: crpAppliedFailedConditions(crp.Generation),
+					PlacementStatuses: []placementv1beta1.ResourcePlacementStatus{
+						{
+							ClusterName: memberCluster1EastProdName,
+							FailedPlacements: []placementv1beta1.FailedResourcePlacement{
+								{
+									ResourceIdentifier: placementv1beta1.ResourceIdentifier{
+										Group:     "dummy",
+										Version:   "v10",
+										Kind:      "Fake",
+										Namespace: workNamespaceName,
+										Name:      wrappedCMName1,
+										Envelope: &placementv1beta1.EnvelopeIdentifier{
+											Name:      wrapperCMName,
+											Namespace: workNamespaceName,
+											Type:      placementv1beta1.ConfigMapEnvelopeType,
+										},
+									},
+									Condition: metav1.Condition{
+										Type:               placementv1beta1.WorkConditionTypeApplied,
+										Status:             metav1.ConditionFalse,
+										Reason:             string(workapplier.ManifestProcessingApplyResultTypeDecodingErred),
+										ObservedGeneration: 0,
+									},
+								},
+							},
+							Conditions: resourcePlacementApplyFailedConditions(crp.Generation),
+						},
+					},
+					SelectedResources: []placementv1beta1.ResourceIdentifier{
+						{
+							Kind:    "Namespace",
+							Name:    workNamespaceName,
+							Version: "v1",
+						},
+						{
+							Kind:      "ConfigMap",
+							Name:      wrapperCMName,
+							Version:   "v1",
+							Namespace: workNamespaceName,
+						},
+					},
+					ObservedResourceIndex: "0",
+				}
+				if diff := cmp.Diff(crp.Status, wantStatus, crpStatusCmpOptions...); diff != "" {
+					return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+				}
+				return nil
+			}, eventuallyDuration*3, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
```

Comment:
any reason we need eventuallyDuration*3?

Reply:
Yeah, it should work fine without; let me drop this.